tests/zarf.yaml

```
@@ -9,29 +9,53 @@ metadata:
   version: 0.1.0
 
 components:
-  - name: test-valkey-connectivity
+  - name: test-valkey-instances
     required: true
     manifests:
-      - name: valkey-cli
-        namespace: valkey-cli
+      - name: valkey-test
+        namespace: valkey-test
         files:
           - valkey/uds-package.yaml
-          - valkey/valkey-cli.yaml
+          - valkey/test-job.yaml
     images:
       - docker.io/bitnami/valkey:8.0.1-debian-12-r2
     actions:
       onDeploy:
+        before:
+          - description: Wait for Valkey-standalone to be healthy
+            maxTotalSeconds: 300
+            wait:
+              cluster:
+                kind: packages.uds.dev
+                name: valkey
+                condition: "'{.status.phase}'=Ready"
+                namespace: valkey-standalone
+          - description: Wait for Valkey-standalone to be ready
+            maxTotalSeconds: 90
+            wait:
+              cluster:
+                kind: pod
+                name: app.kubernetes.io/name=valkey
+                condition: Ready
+                namespace: valkey-standalone
+          - description: Wait for Valkey-replicated to be healthy
+            maxTotalSeconds: 300
+            wait:
+              cluster:
+                kind: packages.uds.dev
+                name: valkey
+                condition: "'{.status.phase}'=Ready"
+                namespace: valkey-replicated-w-sentinel
+          - description: Wait for Valkey-replicated to be ready
+            maxTotalSeconds: 90
+            wait:
+              cluster:
+                kind: pod
+                name: app.kubernetes.io/name=valkey
+                condition: Ready
+                namespace: valkey-replicated-w-sentinel
+          - description: Delete old jobs if they exist
+            cmd: uds zarf tools kubectl delete -f tests/valkey/test-job.yaml || true
         after:
-          - cmd: ./zarf tools kubectl get secret -n valkey-cli valkey --template='{{ index .data "valkey-password" }}' | base64 -d
-            mute: true
-            setVariables:
-              - name: VALKEY_PASSWORD
-                sensitive: true
-          - cmd: |
-              ./zarf tools kubectl exec -n valkey-cli valkey-cli -c valkey-cli -- \
-                sh -c "echo \"ping\\nset foo bar\\nget foo\" | valkey-cli -h valkey-master.valkey.svc.cluster.local -a ${ZARF_VAR_VALKEY_PASSWORD}"
-            setVariables:
-              - name: VALKEY_OUTPUT
-          - cmd: echo ${ZARF_VAR_VALKEY_OUTPUT} | grep PONG
-          - cmd: echo ${ZARF_VAR_VALKEY_OUTPUT} | grep OK
-          - cmd: echo ${ZARF_VAR_VALKEY_OUTPUT} | grep bar
+          - description: Watch test jobs and report their conditions
+            cmd: /bin/sh tests/watch-jobs.sh
```

```suggestion
            cmd: ./tests/watch-jobs.sh
            shell:
              linux: bash
              darwin: bash
```
the script uses bash-specific array syntax so needs shell preferences to work correctly.

---

I love this - didn't know GH well enough to know I could do that so i removed the array and went to posix shell compliance.

---

Oh, that's Zarf anyway.

---

sorry I should have dig more - this shouldn't actually be necessary - the real reason it wouldn't work was that the script was not executable at first.